### PR TITLE
#390: Replace all usages of Arrays.asList with List.of.

### PR DESCRIPTION
**Pull request by rdvdijk**
This is a minor clean-up: Replace all usages of `Arrays.asList` with the more modern (Java 9 :laughing:) `List.of`.

### CODE REVIEW

File: core/src/test/java/io/parsingdata/metal/ErrorsTest.java [hunk 1]
```diff
@@ -23,7 +23,6 @@
 import static io.parsingdata.metal.Shorthand.con;
 import static io.parsingdata.metal.Shorthand.def;
 import static io.parsingdata.metal.Shorthand.div;
-import static io.parsingdata.metal.Shorthand.last;
```
Comment (rdvdijk) [inline core/src/test/java/io/parsingdata/metal/ErrorsTest.java:26]: Bonus: Removed an unused import.

File: core/src/test/java/io/parsingdata/metal/expression/value/RefEdgeCaseTest.java [hunk 2]
```diff
@@ -20,7 +20,6 @@
 
 import static io.parsingdata.metal.Shorthand.con;
 import static io.parsingdata.metal.Shorthand.div;
-import static io.parsingdata.metal.Shorthand.exp;
```
Comment (rdvdijk) [inline core/src/test/java/io/parsingdata/metal/expression/value/RefEdgeCaseTest.java:23]: Bonus: Another unused import.

File: core/src/test/java/io/parsingdata/metal/util/TokenDefinitions.java [hunk 3]
```diff
@@ -19,13 +19,11 @@
 import static io.parsingdata.metal.Shorthand.con;
 import static io.parsingdata.metal.Shorthand.def;
 import static io.parsingdata.metal.Shorthand.div;
-import static io.parsingdata.metal.Shorthand.last;
 import static io.parsingdata.metal.Shorthand.not;
 import static io.parsingdata.metal.Shorthand.ref;
 
 import io.parsingdata.metal.Shorthand;
 import io.parsingdata.metal.expression.value.SingleValueExpression;
-import io.parsingdata.metal.expression.value.ValueExpression;
```
Comment (rdvdijk) [inline core/src/test/java/io/parsingdata/metal/util/TokenDefinitions.java:28]: Bonus: Two more unused imports.

File: core/src/test/java/io/parsingdata/metal/BitwiseValueExpressionSemanticsTest.java [hunk 4]
```diff
@@ -41,7 +41,7 @@ public class BitwiseValueExpressionSemanticsTest extends ParameterizedParse {
 
     @Parameters(name = "{0} ({4})")
     public static Collection<Object[]> data() {
-        return Arrays.asList(new Object[][] {
+        return List.of(new Object[][] {
```
Comment (mvanaken) [inline core/src/test/java/io/parsingdata/metal/BitwiseValueExpressionSemanticsTest.java:44]: I have been itching to convert these parameterized test using Stream<Arguments>, and get rid of Junit4 altogether, but that is a different issue. ;) Shall I create a new issue for this?

**Review by mvanaken — APPROVED** [review]
An other step forward! :)

File: core/src/test/java/io/parsingdata/metal/BitwiseValueExpressionSemanticsTest.java (hunk 4 above)
Comment (Carly-1) [reply core/src/test/java/io/parsingdata/metal/BitwiseValueExpressionSemanticsTest.java:44]: Sounds like a good plan! :)

**Review by Carly-1 — APPROVED** [review]

**Pull request merged**